sherpa/ui/utils.py

```diff
@@ -2288,8 +2340,29 @@ def get_data(self, id=None):
         """
         return self._get_item(id, self._data, 'data set', 'has not been set')
 
+    def _get_data(self, id):
+        """Return a data set or None.
+
+        The same as get_data except that it returns None if the
+        dataset does not exist.
+
+        Parameters
+        ----------
+        id : int or str, optional
+           The data set. If not given then the default
+           identifier is used, as returned by `get_default_id`.
+
+        Returns
+        -------
+        instance : sherpa.data.Data or None
+
+        """
+
+        return self._data.get(self._fix_id(id))
```

Comment:
This is a very-simple routine, and I feel a bit bad about adding it as a method, but it does (I believe) make the downstream code a bit cleared - i.e. we call `self.get_data` if we require the data to exist and `self._get_data` if the data need not exist.